src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java

```diff
@@ -973,34 +974,27 @@ public String getCastScriptStatement(String name, String castType, List<KVValue>
      * approach will return type of result column as DOUBLE, although there is enough information to understand that
      * it might be safely treated as INTEGER.
      */
-    public static Schema.Type getScriptFunctionReturnType(Field field) {
+    public static Schema.Type getScriptFunctionReturnType(
+            int fieldIndex, MethodField field, ColumnTypeProvider scriptColumnType) {
```

Comment:
The input argument could be reduced as (MethodFiled field, Schema.Type resolvedType)

Reply:
good catch - will change